app/league/configuration/form/form.tsx

```diff
@@ -1,67 +1,100 @@
 import Select from "@/app/components/baseComponents/components/inputs/select/select";
 import Button from "@/app/components/baseComponents/components/button/button";
 export default function LeagueConfigurationForm(){
-    const formSubmit = (e: MouseEvent<HTMLButtonElement>)=> { e.preventDefault(); };
-    return (
-        <form className={styles.form}>
-            <TextInput
-                label="Wikipedia Page Name"
-                placeholder="example: big_brother"
-                isRequired={true}
-                validationPattern={validationPattern.wikiPageName.string}
-                id="wikiPageUrl"
-            />
-            <TextInput
-                label="Wiki Section Header"
-                placeholder="example: HouseGuests"
-                isRequired={true}
-                validationPattern={validationPattern.wikiSectionHeader.string}
-                id="castPhrase"
-            />
-            <TextInput
-                label="League Key Prefix"
-                placeholder="example: big_brother:27"
-                isRequired={true}
-                validationPattern={validationPattern.leagueKey.string}
-                id="leagueKeyPrefix"
-            />
-            <TextInput
-                label="Contestant Type"
-                placeholder="example: team"
-                isRequired={true}
-                validationPattern={validationPattern.contestantType.string}
-                id="contestantType"
-            />
-            <Select 
-                labelText="League Status"
-                selectOptions={
-                    [{
-                        value: "active",
-                        text: "Active"
-                    },{
-                        value: "archived",
-                        text: "Archived"
-                    }]
+    // This is needed to allow for query selector below
+    const formRef = useRef(null as HTMLFormElement | null);
+    const [errorMsg, setErrorMsg] = useState("");
+    const router = useRouter();
+    const formSubmit = async (e: MouseEvent)=> { 
+        e.preventDefault();
+        if(formRef.current){
+            const formData = new FormData(formRef.current);
+            const formObject = Object.fromEntries(formData);
+            const formDataAsJson = JSON.stringify(formObject);
+            try{
+                const response = await fetch("/api/league", {
+                    method: "POST",
+                    headers: {
+                        "Content-Type": "application/json"
+                    },
+                    body: formDataAsJson
+                });
+                if(!response.ok){
+                    setErrorMsg("There was a problem. Please check the form and try to resubmit.");
+                }
+                const result = await response.json();
+                if(result.message === "posted"){
+                    router.push("/");
```

Comment:
I see all sorts of untested code here, right?

Reply:
Yep. I'm pushing those tests up now. I forgot to commit them

Reply:
Update: They are up here now :)